src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.MovieSessionDao;
+import com.cinema.exception.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.MovieSession;
+import com.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session!", exception);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<MovieSession> getAvailableSessions(Long movieId, LocalDate showTime) {
```

Comment:
```suggestion
    public List<MovieSession> getAvailableSessions(Long movieId, LocalDate date) {
```
I think it's better to say 'date' and not 'showTime' here because you don't actually get time in hours:mins as a parameter, and the name 'showTime' could be confusing

Reply:
> I think it's better to say 'date' and not 'showTime' here because you don't actually get time in hours:mins as a parameter, and the name 'showTime' could be confusing

\+